ArmoniK.SDK.Client/src/SessionServiceImpl.cpp

```diff
@@ -351,51 +301,40 @@ void SessionServiceImpl::CleanupTasks(const std::set<std::string> &task_ids) {
   // Cancel the given tasks
   while (tasks_iterator != task_ids.end()) {
     channel_pool.WithChannel([&](const std::shared_ptr<::grpc::Channel> &channel) {
-      auto stub = armonik::api::grpc::v1::tasks::Tasks::NewStub(channel);
-      ::grpc::ClientContext context;
-      armonik::api::grpc::v1::tasks::CancelTasksRequest request;
+      std::vector<std::string> batched_ids;
       for (size_t i = 0; i < batch_size && tasks_iterator != task_ids.end(); ++i) {
-        *request.mutable_task_ids()->Add() = *tasks_iterator;
+        batched_ids.push_back(*tasks_iterator);
```

Comment:
First is better intent:
- `push_back` to add an element that already exist (either copy or move)
- `emplace_back` to construct an element in situ

Second is actually safety, `push_back` takes only `const X&` and `X&&` and is guaranteed to call either the copy or the move constructor, nothing else. While `emplace_back` can take anything and call any constructor, even ones marked `explicit`. This might be what you want, but if you know that your element already exist, it is not what you want.

Third, it `push_back` plays nicer with argument deduction, and allows you to write code like that:
```cpp
struct Coord {
	int x;
	int y;
};

void add_coord(std::vector<Coord>& coords, int x, int y) {
	coords.push_back({x, y});
}
```

Also, I should highlight that for copy and move, the generated code is actually identical with both `push_back` and `emplace_back`

More information: https://stackoverflow.com/questions/10890653/why-would-i-ever-use-push-back-instead-of-emplace-back/36919571#36919571